lib/src/screens/connect_device/monero_hardware_wallet_options_page.dart

```diff
@@ -0,0 +1,220 @@
+                                  });
+                                },
+                                icon: Container(
+                                  padding: const EdgeInsets.all(8),
+                                  decoration: BoxDecoration(
+                                    borderRadius: BorderRadius.circular(6.0),
+                                    color: Theme.of(context).hintColor,
+                                  ),
+                                  width: 34,
+                                  height: 34,
+                                  child: Image.asset(
+                                    'assets/images/refresh_icon.png',
+                                    color: Theme.of(context)
+                                        .extension<SendPageTheme>()!
+                                        .textFieldButtonIconColor,
+                                  ),
+                                ),
+                              ),
+                            ),
+                          ),
+                          validator: WalletNameValidator(),
+                        ),
+                      ],
+                    ),
+                  ),
+                ),
+                Padding(
+                  padding: EdgeInsets.only(top: 20),
+                  child: BlockchainHeightWidget(
+                    focusNode: _blockHeightFocusNode,
+                    key: _blockchainHeightKey,
+                    hasDatePicker: true,
+                    walletType: WalletType.monero,
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+        bottomSectionPadding: EdgeInsets.all(24),
+        bottomSection: Observer(
+          builder: (context) {
+            return LoadingPrimaryButton(
+              onPressed: _confirmForm,
+              text: S.of(context).seed_language_next,
+              color: Colors.green,
+              textColor: Colors.white,
+              isDisabled: _walletHardwareRestoreVM.name.isEmpty,
+            );
+          },
+        ),
+      ),
+    );
+  }
+
+  Future<void> _confirmForm() async {
+      showPopUp<void>(
+        context: context,
+        builder: (BuildContext context) => AlertWithOneAction(
+            alertTitle: S.of(context).proceed_on_device,
+            alertContent: S.of(context).proceed_on_device_description,
+            buttonText: S.of(context).cancel,
+            buttonAction: () => Navigator.of(context).pop()),
+      );
+
+    final options = {};
+    options['height'] = _blockchainHeightKey.currentState?.height ?? -1;
+    await _walletHardwareRestoreVM.create(options: options);
+  }
+
+  bool _effectsInstalled = false;
+
+  void _setEffects(BuildContext context) {
+    if (_effectsInstalled) return;
+
+    reaction((_) => _walletHardwareRestoreVM.error, (String? error) {
+      if (error != null) {
+        if (error == S.current.ledger_connection_error) Navigator.of(context).pop();
```

Comment:
why do we just pop? is it displayed to the user somewhere else?

Reply:
It is shown right after so we co back to the connection screen and then we show the popup. Navigator.pop() is not a return.